File: BOJ_2023/2309_일곱난쟁이/main.cpp
// https://www.acmicpc.net/problem/2309

#include <bits/stdc++.h>
using namespace std;

int arr[9];
vector<int> answer;
void combi(int start, int n, int r, vector<int> v) {
  if (r == v.size()) {
    int sum = 0;
    for (auto i : v) sum += i;
    if (sum == 100) answer = v;
    return;
  }

  for (int i = start + 1; i < n; i++) {
    v.push_back(arr[i]);
    combi(i, n, r, v);
    v.pop_back();
  };
}

int main() {
  for (int i = 0; i < 9; i++) {
    cin >> arr[i];
  }
  vector<int> v;
  combi(-1, sizeof(arr) / sizeof(int), 7, v);
  sort(answer.begin(), answer.end());
  for (auto i : answer) cout << i << "\n";
  return 0;
}

File: BOJ_2023/2309_일곱난쟁이/main.js
// https://www.acmicpc.net/problem/2309

const readline = require('readline')
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
})

let input = []

rl.on('line', (line) => {
  input.push(parseInt(line))
}).on('close', () => {
  solve(input)
  process.exit()
})

function solve(input) {
  const answer = []
  const total = input.reduce((p, c) => p + c)
  for (let i = 0; i < input.length; i++) {
    for (let j = 0; j < i; j++) {
      const sum = total - input[i] - input[j]
      if (sum !== 100) continue
      for (let k = 0; k < input.length; k++) {
        if (k == i || k == j) continue
        answer.push(input[k])
      }
    }
    if (answer.length !== 0) break
  }
  console.log(answer.sort((a, b) => a - b).join(' '))
}
